pkg/ansible/watches/watches.go

```diff
@@ -272,16 +272,13 @@ func getMaxWorkers(gvk schema.GroupVersionKind, defValue int) int {
 		"_",
 		-1,
 	))
-	maxWorkers, err := strconv.Atoi(os.Getenv(envVar))
-	if err != nil {
-		// we don't care why we couldn't parse it just use default
-		log.Info("Failed to parse %v from environment. Using default %v", envVar, defValue)
-		return defValue
-	}
-
-	if maxWorkers <= 0 {
-		log.Info("Value %v not valid. Using default %v", maxWorkers, defValue)
-		return defValue
+	maxWorkers := defValue
+	if val, ok := os.LookupEnv(envVar); ok {
+		if i, err := strconv.Atoi(val); err != nil {
+			log.Info(fmt.Sprintf("Unable to find a value for %v. Using default value for maxWorkers %d", envVar, defValue))
```

Comment:
I could be wrong, but I think @camilamacedo86 was suggesting that we should log a separate message if we use the default when the environment variable is not set. I would also suggest we follow the [structured logging conventions of the logr.Logger](https://godoc.org/github.com/go-logr/logr#hdr-Parts_of_a_log_line).

So here, we can use the following since we have encountered an error trying to parse the string as an integer.
```go
log.Info("could not parse environment variable as an integer; using default value", "envVar", envVar, "default", defValue)
```

For @camilamacedo86's suggestion, if we do the `os.LookupEnv()` and `ok == false`, that means the environment variable was not set. In think in _that_ case, we could log this message:
```go
log.Info("environment variable not set; using default value", "envVar", envVar, "default", defValue)
```

@camilamacedo86 @jmrodri @bharathi-tenneti WDYT?